Campaigns: Add Wings of Victory (Drake Campaign)

sigurdfdragon opened the pull request:
Will leave this up for at least 2 weeks.

The contents of this PR match the current state of:
https://github.com/sigurdfdragon/Wings_of_Victory/tree/1.15
and will be updated as needed.

The current version of the campaign on the 1.14 add-on server is here:
https://github.com/sigurdfdragon/Wings_of_Victory

Forum Thread:
https://forums.wesnoth.org/viewtopic.php?f=8&t=49140

I left the version number in for now. Would prefer to leave it in during the 1.15 cycle to help keep track of changes ported from the 1.14 add-on version.

nemaara commented:
I have a few reservations about adding drakes consuming human meat as a point of lore (and, for example, shadowm also agrees with me). Nowhere else in mainline is there even an inkling of this being a thing, so is there any way you can change it? 

As far as I can see, most of the scenarios past the first couple are more about Galun/the Drakes treating others as lesser races (but not strictly as eating them as food, only that there is a shortage of unspecified food). If you can perhaps alter the dialogue in the first couple scenarios to have the Drakes have a different food source, I think it would be preferable.

A more concrete example: S1 is a hunt on humans, but it could just be that the humans are kept as slaves (not food), and some slaves may be trying to escape. Alternatively, hunting down slaves could simply be a sport for Drakes, much like having gladiator fights was a thing for the Romans.

CelticMinstrel commented:
You can add me to the list of people who see that as a questionable narrative choice.

sigurdfdragon commented:
At the start of the campaign, they feed on a wide variety of meat (including human), since they see themselves as a descendant from an apex predator. At the end of the campaign, human and some other meats are off their menu, due to the realities of living on the great continent. (ie, some of the talking meat is dangerous in numbers)
Flying carnivores don't make a thing of wasting good meat, so they are eating what they are running across during the rest of the journey, it's just not front and center.

This seems fine to me.
Is it just the eating human part that you two are concerned about? (ie, orcs & trolls are fine being on the menu)

CelticMinstrel commented:
No, in this context I'd consider humans and orcs (and elves and dwarves and saurians) to be equivalent. I'd also consider trolls to be equivalent, but I can also see additional possible reasons to keep them off the menu (aren't they a rock-based life form?).

It really depends on the details, so I'd have to actually read through the campaign before I can say anything more than just "sounds questionable". It's certainly not impossible to justify this choice. For example, if you were intending to depict drakes as adhering to an alternate system of morality (in the sense of fae - what TVTropes refers to as "blue and orange morality"), then it could be acceptable. (I don't personally see the drakes as fitting that kind of profile, mind you.)

A more concrete possibility that might be a little less questionable (though still depends on other details) would be that they'll eat any meat that comes their way, but they don't intentionally hunt intelligent species - so any humans or orcs they eat would either be already dead or were killed purely in self-defence.

The first scenario as described by nemaara is particularly questionable. From that description, it sounds like they are intentionally and knowingly hunting intelligent species for food. From the start, that pretty much makes them sound like the villain. If that's what you're going for, maybe it's okay, but if you're trying to make them more obviously sympathetic, I wouldn't really recommend that path.

I'd also add that as a general food source, humans or orcs or other similar species wouldn't really be worth the risk compared to animals such as cows, horses, or taurochs, which is the kind of thing I'd expect drakes to rely on as their primary food source. So there are practical reasons to avoid it, in addition to moral reasons. Now, if the primary source has died out or migrated away, and they're desperate for food and going reluctantly after humans as a last resource, well, that _might_ make them seem less villainous.

I'll try to read through the first two or three scenarios later this week and give a more specific opinion rather than these vague thoughts,

sigurdfdragon commented:
I definitely am aiming to depict drakes to have a different culture than loyalists or merfolk, so some flavors of blue/orange morality may very well end up with it.

They hunt humans in Morogor at the start, because there, they are apex predators and nothing poses a real challenge to them. This changes as Galun's tribe journeys from Morogor to the Great Continent. I'm not going for obvious sympathetic here, at least to start in the campaign.

> I'd also add that as a general food source, humans or orcs or other similar species wouldn't really be worth the risk compared to animals such as cows, horses, or taurochs...

Part of what's being aimed for in WoV is that they realize this during the journey. And I want them to have more of a practically slant behind their actions compared to moral (but still have some moral). "A lion does not care about the opinions of sheep" type thing.

sigurdfdragon commented:
Updated the PR with changes from version 0.9.5

CelticMinstrel commented:
Argh, I forgot to get to this...

Maybe I can still find time tomorrow or the next day? If not it'll have to wait till next weekend...

jostephd commented on data/campaigns/Wings_of_Victory/scenarios/05_Islands.cfg:
```diff
@@ -0,0 +1,336 @@
+            {VANK}
+            placement=leader
+        [/unit]
+    [/side]
```
When I load the S5 turn 1 autosave, it says "It is now josteph’s turn", whereas for say HttT it says "It is now Konrad’s turn", so I expected it to say "It is now Galun’s turn" here. I guess `side_name=_ "Galun"` should be added to the [side] tag?

Or maybe the engine should always try a [side][leader]name= key if there's no [side]name= key.

CelticMinstrel replied:
Yeah, the engine should probably do that.

sigurdfdragon replied:
Noted

CelticMinstrel commented on data/campaigns/Wings_of_Victory/_main.cfg:
```diff
@@ -0,0 +1,70 @@
+[/units]
+#endif
+
+# TODO: use em dashes & unicode quote? Pango doesn't seem to like it on my system,
```
Could this be because you haven't told your editor to save as UTF-8?

sigurdfdragon replied:
Will give this a try, thanks.

CelticMinstrel commented on data/campaigns/Wings_of_Victory/scenarios/01_The_Hunt.cfg:
```diff
@@ -0,0 +1,450 @@
+            rabbit_number=5
+            rabbit_enemy_distance=3
+            ## rabbit_hole_img=scenery/burrow.png # commented out due to bug in forest_animals mai, can't use any other [item] unless I'm fine with rabbits coming out of them until it is fixed.
+            ## Can remove comments and activate for 1.15.0, leave in 1.14.x, as it was only fixed in 1.14.6
```
If I understand the meaning here correctly, this says the bug was fixed, right? So it should be uncommented.

sigurdfdragon replied:
will update in next push.

CelticMinstrel commented on data/campaigns/Wings_of_Victory/scenarios/01_The_Hunt.cfg:
```diff
@@ -0,0 +1,450 @@
+                condition=lose
+            [/objective]
+            [objective]
+                description= _ "All enemies are defeated and Galun has not reached level two."
```
This feels like an unfair loss condition. You defeat all your enemies, and then lose. It would make more sense to me to just spawn more enemies if Galun is not yet level 2... though I guess that might undermine the premise a little. Still... why does he absolutely need to be level 2 anyway?

sigurdfdragon replied:
It's part of the backstory of the drakes. Short version is that they are not fully mature until they are level 2.

As for the mechanics, it is difficult to defeat all the enemies and not make the objective, and even moreso since the addition of rabbits as they respawn automatically.
Maybe I could revise that somehow...

jostephd replied:
I playtested S1 on 0.9.5 a few days ago and advanced Galun on turn 12 without too much trouble (I actually finished the scenario with Galun being lvl2 with 43/64XP). That objective makes for varied gameplay, has a lore justification, and isn't unreasonably difficult...I don't see a problem with keeping it.

CelticMinstrel commented on data/campaigns/Wings_of_Victory/scenarios/01_The_Hunt.cfg:
```diff
@@ -0,0 +1,450 @@
+                condition=lose
+            [/objective]
+            [objective]
+                description= _ "You lose more than one member of your hunting party."
```
The phrasing of this objective doesn't match that of the standard objectives. Something like "Loss of two members of your hunting party" would fit better.

The other two objectives below also suffer from this problem, but I'm not sure how they can be fixed, and I think they have other problems too, so for now I won't make any suggestions.

sigurdfdragon replied:
Noted.

CelticMinstrel commented on data/campaigns/Wings_of_Victory/scenarios/01_The_Hunt.cfg:
```diff
@@ -0,0 +1,450 @@
+        [/message]
+        [message]
+            speaker=Galun
+            message= _ "Speaking of the hunt, let's take those runners there. You must been distracted or blind yourself, not spotting them before."
```
"must have been"

sigurdfdragon replied:
Noted

CelticMinstrel commented on data/campaigns/Wings_of_Victory/scenarios/01_The_Hunt.cfg:
```diff
@@ -0,0 +1,450 @@
+
+        [message]
+            speaker=Gerth
+            message=_ "The damned worms have found us! Send the women and children to the caves. To arms, men."
```
"worms" or "wyrms" :P

(Not saying "wyrms" _is_ appropriate, but ... it _might_ be?)

sigurdfdragon replied:
Yes, it would be. Thanks.

shikadiqueen commented:
Regarding the rabbits in Invasion from the Unknown, they are from this post:
https://forums.wesnoth.org/viewtopic.php?p=588729#p588729

sigurdfdragon commented:
Will update the incisors with the fangs-rodent.png
Thanks for the better rabbit images link

Will think about what to do with the terms and the hunting.

slavrenyuk commented:
Human hunting is definitely inappropriate for young players. Though I like the idea that drakes originally hunt and eat people, but than intentionally stopped to do that. In my opinion, if you show the worst and the best sides of a character (or a whole race in this case) it makes them very deep and alive.  

Some ideas to mitigate the human hunting theme. 

Do not focus on that. Give some hints recognizable by adults, but not obvious for younger players. 

Say they are not cruel, but different. They don't want to make people suffer. Because of their warlike nature they may truly believe that it is better to die young in a battle rather than live long and peaceful life.

shikadiqueen commented:
> Human hunting is definitely inappropriate for young players.

At least two campaigns casually feature genocide (LoW and UtBS). The game is supposed to be roughly PG 13 as a whole.

Personally I'm more bothered with trying to shoehorn race characterization into the game using only one particular campaign, like @nemaara said earlier. That can only result in a more inconsistent bigger picture for mainline, bearing in mind that they're lesser one-off antagonists in TRoW, a hidden scenario in HttT, and ascended to **allies** in NR.

slavrenyuk commented:
If drakes stopped to eat humans just because humans are dangerous, that depicts drakes more like an animals. In that case they are not deep, but instead flat characters. I am not sure that most players will like to play an animal role. That is probably ok for a single campaign, but not as a characteristic of a core race.

What I would like to see. Initially drakes were like an animals. But then they evolved. Some of them decided to not eat people because of the morality. While other just don't want to mess with the strong enemy and prefer to hunt an easier prey.

CelticMinstrel commented:
> I definitely am aiming to depict drakes to have a different culture than loyalists or merfolk, so some flavors of blue/orange morality may very well end up with it.

I want to add that a blue and orange morality system isn't really needed just to depict an alternate culture; usually it suggests a species with a very different thought process. In Wesnoth, perhaps wose or trolls could fit this definition, as well as fairies or djinn. I'm not saying drakes _couldn't_ be like this, but as far as I know, they're normal flesh-and-blood creatures just like humans, elves, dwarves, or saurians, so it would be kind of strange for them to have such a different thought process.

> What I would like to see. Initially drakes were like an animals. But then they evolved.

The problem with this idea is ordering. The evolution you refer to probably predates language, so it doesn't make sense to be depicting it in this campaign. Of course, humans underwent a similar evolution, but you don't see any campaigns trying to depict _that_, right?

jostephd commented:
I haven't played all of mainline, but I don't see what the problem is with saying drakes used to eat humans on Morogor (and I haven't seen much explanation of the problem in the comments, either). We're in the realm of fantasy and the concept of a race of intelligent predators that eat their enemies after killing them doesn't shake my disbelief the slightest bit. Humans on Earth are an intelligent race that (broadly speaking) doesn't eat its enemies, but when we're in a fantasy or science fiction setting, I can readily imagine intelligent races that don't share our taboo on cannibalism.

That said, I do agree with @shikadiqueen that we should be mindful of keeping mainline's universe consistent across campaigns.

CelticMinstrel commented:
> the concept of a race of intelligent predators that eat their enemies after killing them doesn't shake my disbelief the slightest bit.

That's not the impression I got from the first scenario at all. They were deliberately hunting humans, and furthermore, claimed to sometimes deliberately let slaves escape so they could hunt them. That's a far cry from going to war with humans and eating their enemy's fallen when they win. Mind you, I find the latter questionable too, but... less so than what's actually portrayed in that scenario.

jostephd commented:
> That's not the impression I got from the first scenario at all. They were deliberately hunting humans, and furthermore, claimed to sometimes deliberately let slaves escape so they could hunt them.

The line is ["Indeed, but runners are getting hard to find, even though we do let some of our slaves escape to stock the ranges. Too many hunters, not enough game."](https://github.com/sigurdfdragon/Wings_of_Victory/blob/3bd499ea2b481710e27f8ca8429941e0d40d51f5/scenarios/01_The_Hunt.cfg#L439)

I didn't read as "Let's release the slaves, give them a five minutes' head start, and then hunt them for sport". I read it as an ecosystem crisis: the drake population had grown beyond what the islands can sustain, so they preyed on humans to the point that humans are endangered / near extinction within the drakes' territory, and if humans go completely extinct the food chain will collapse, so the drakes let some humans escape...but there's still population pressure on the drakes' side, which ultimately drives Galun to leave Morogor. (The last part is recounted later in the campaign.)

This ecosystem crisis interpretation would also explain why there's [an infinite amount of rabbits](https://github.com/wesnoth/wesnoth/pull/3977#discussion_r268376153) :grin:

slavrenyuk commented:
> The problem with this idea is ordering. The evolution you refer to probably predates language, so it doesn't make sense to be depicting it in this campaign. Of course, humans underwent a similar evolution, but you don't see any campaigns trying to depict that, right?

It reminds me vegetarians and is not related to language. Their biology haven't changed, but their mentality is different. That is what I meant. And there is a campaign describing an evolution of elves. 

I understand that "drakes eat humans" may not fit the mainline. But if it will be somehow added to the mainline,  the game would be more fun to me.

CelticMinstrel commented:
It's not really the same as members of a meat-eating species deciding to become vegetarian. It's more like members of a cannibalistic species deciding to stop eating their own kind. Sure, it's not _strictly_ cannibalism in this case, but it's pretty close.

slavrenyuk commented:
Perhaps you are right. Should we notify the community about this discussion? I know that @beetlenaut created several mainline campaigns, it is interesting to hear his opinion. @sigurdfdragon , do you still want to depict drakes like an animals with minimum morale? Hope you will come to a consensus. Having drakes campaign in the mainline would be a great improvement.

CelticMinstrel commented:
Note: "morale" and "morals" mean completely different things; I think you meant "morals" here. I'd also add that "morals" might not really apply to animals, but on the other hand, depicting a sentient species as animals is a terrible narrative choice.

slavrenyuk commented:
Thanks for corrections @CelticMinstrel. Yes, I meant "morals". Updated my comment.

CelticMinstrel commented:
If the intention is really to have drakes eat humans, I think dropping the slave angle would be a good start. As it is now, it just sounds like they're hunting them for sport, and honestly, what do drakes need human slaves for anyway? Why not just have them come into conflict with humans over limited resources? Or you could just drop the humans entirely from the first scenario, as they're not strictly necessary to the goal of levelling up your leader.

jostephd commented:
> what do drakes need human slaves for anyway?

According to drake race description, they are predators but cultivate fungi in caverns. Maybe humans are better than drakes at farming fungi, or for that matter at raising cows? The drake population has grown to the point that the islands can't sustain them without the humans' farming. The drakes used to be parasites of the humans and raid their fields, but now out of necessity they enslaved some humans so they can consume _all_ output of those humans' fields.

Or maybe they just have the human slaves do the most dangerous jobs, like standing guard in cavern entrances in case cave-dwelling monsters come out of the mountain?

Or maybe the relationship between drakes and humans is like a feudal lord and vassal, the humans work for the drakes and in exchange the drakes protect the humans from others? (Then, when the population grows to the point of resource shortage, some humans try to overthrow the drakes, and that's why in S1 the drakes and humans fight?)

Also, in general maybe drakes gain some advantage from (unslaved) humans being on the islands. For example, maybe the island is occasionally raided by nagas, and the (unslaved) humans, nagas, and drakes are in a three-way war. This would mean drakes have an interest in having enough unslaved humans on the islands that they (the humans) will be able to pose a threat to the nagas, in order for the humans and nagas to deplete their forces fighting each other. Then, when the human population decreases, the drakes free slaves in order to assist the humans against the nagas (the enemy of my enemy is my friend).

CelticMinstrel commented:
For version 1, I really want to know what they need the fungi for. To me, the most obvious reason is somehow a fuel for their fire, but the lore clearly states their fire doesn't require fuel, so that doesn't really work. Is it a dietary supplement? A psychedelic herb? Some kind of ritual? I could definitely see humans being better than drakes at raising cows, though; I imagine the cows would be pretty uneasy in the presence of drakes. (Cows are probably one of the drakes' primary food sources, I would think.)

For version 3, that doesn't sound like slavery to me, though I suppose one can argue that feudal serfs weren't that much different from slaves.

As for version 2, I think drakes would feel that they're better suited for that particular job, at least, so it doesn't work for me. I suppose maybe if humans have better vision than drakes, it _might_ work, but even then...

In your final paragraph, you keep talking about "unslaved" humans, meaning free humans? Or did you actually mean to say _"enslaved"_?

jostephd commented:
> For version 3, that doesn't sound like slavery to me, though I suppose one can argue that feudal serfs weren't that much different from slaves.

Why should it sound like slavery? If we're changing the scenario anyway we don't have to keep the slavery part. We'd say that drakes and humans lived a stable but unequal coexistence (humans being serfs of the drakes) and that when the drakes migrated to the Great Continent they decided not to provoke the local humans who were militarily superior to the drakes.

I hope this idea won't make it too difficult to write a prequel campaign, if we ever want to do that.

> In your final paragraph, you keep talking about "unslaved" humans, meaning free humans? Or did you actually mean to say _"enslaved"_?

Yes, I meant free humans.

CelticMinstrel commented:
> Why should it sound like slavery?

Well, you offered it as an answer to "why would drakes need human slaves", so obviously the implication would be that you're intending to imply it to be a form of slavery.

But yeah, of course, we don't need to keep the slavery part.

Pentarctagon commented:
My two cents - the slavery part doesn't really bother me (story-wise).  Given that humans are smaller, their fingers are probably more dextrous than the Drakes' claws, and would be less likely to spook animals like cows provides multiple other reasons for why they would be kept as slaves by the Drakes, in addition to the Drakes using the humans for menial tasks and-or perhaps certain more dangerous ones.

An alternate dialogue could perhaps mention that while they are hungry, their slaves are starving, and soon they may need to start doing X, Y, and Z tasks themselves.

sigurdfdragon commented:
What do dragons eat?
Would anyone here seriously argue that we should have depictions of dragons that actively avoid eating humans, orcs, dwarves and whatever else without a good reason?

What do their descendants eat?

I'm aiming for a solid characterization that rings true with the realities of Wesnoth universe.
So I'm inclined to go for some orange/blue (bacon/necktie) morality here. Drakes need to eat.

I'm open to considering having the question of food some degree of less obvious.

They evolve over the course of the campaign to deal with the practicality of some of the meat being significantly more dangerous to hunt on the Great Continent than it was on Morogor. It seems like they would also have some respect for other dangerous predators, once they have realized they are dangerous.

I also want to tie-in with how the drakes are in TRoW, HttT, THoT, DW, & NR. I'd say it's mostly there already, 1-3 more pieces of dialog should suffice.

I'll get to a review of some of the above once I get the changes from the previous week implemented and the PR updated with 0.9.6

CelticMinstrel commented:
I would be inclined to say that any lineal relationship between dragons and drakes is no more than myth. The two species are actually _very_ different, so it doesn't make sense to me for one to evolve from the other. That said, I don't think dragons would normally go after humans and the like. That doesn't mean actively avoiding them, it's just that they're not the best food choice out there – large livestock provide more food and probably fight back less.

Pushing the eating of humans to purely backstory would be less problematic than actually directly depicting it. I think for the most part the problems lie not so much in the content but the way it's depicted. I mean, I do question the idea of them eating humans, but done in a different way (less obvious, or more tactful; and ideally without the slavery, as the mix makes things even worse), perhaps I wouldn't mind it as much.

sigurdfdragon commented:
Updated the PR with version 0.9.6.
See https://github.com/sigurdfdragon/Wings_of_Victory/tree/1.15 for the changes

Thanks for the discussion regarding what the drakes eat, I've gotten some ideas on how to proceed. Hopefully, some adjustments will appear in the next version (0.9.7)

Wedge009 commented:
I've completely missed this until just now but I did see the conversation regarding orcs eating humans (#4007).

Like a few others here I feel uncomfortable with the deliberate eating of sapient races although after reading through the discussion I can understand the 'foreign thinking/culture' depiction attempt. It is good to know that the eating of humans is phased out later in the campaign. I have not played this campaign before nor have I read through the scenario files yet - what time period is it set in? There was some discussion earlier about human dietary habits changing over time and comparing that to the drakes here.

sigurdfdragon commented:
> what time period is it set in?
5-6YW (not too long after TRoW)

Wedge009 commented:
I did a quick play-through of version 0.9.6 of the campaign as available in the add-ons. Maybe, at least in part, because of adjustments after the feedback here the portrayal of drakes eating other races doesn't seem as bad as it sounded earlier. I see two basic reasons: the drakes' mentality that they are superior to other races, and as their numbers have grown more than the island of Morogor can support they need to make do with any meat they can find, even if that means other sapient creatures (not that they care about such 'lesser' races anyway).

A few remarks/questions: How did humans come to be on Morogor? Were they left behind during the flight from the Green Isle? Escapees, deserters and the like?

I am confused what is meant by short vs long pig. I suppose long pig means any humanoid race, not necessarily humans? (Small long pig = dwarf) How does the term runner fit into this? Specifically humans? Specifically the human slaves the drakes kept on Morogor?

The level requirement for the first two campaigns is a bit odd, considering purely from a game-play view. I suppose it fits in with the drake concept of honour and prestige for the 'Contention', but I'm not sure it plays that well. I found a problem too: if Galun's killing blow leading to victory and the end of the scenario is what brings him to the next level, the scenario is still treated as a failure because the level advancement occurs after the ending dialogue. Not sure if there's a way around that, it might be limited by the way the engine orders events.

From the opening of scenario 6:
> some messengers from the other flights left back to Morogor
This doesn't make sense to me - I would write '_flew_ back to Morogor' or '_returned_ to Morogor'.

The scenarios attacking Elensefar (it's not yet the major city that it would become in later years, nice) and Wesmere: I find it curious that there's no dialogue at all between the drakes and the humans/elves. While I suppose humans would be quick to attack anything they feared, I would have thought the elves would make more of an attempt to communicate before resorting to violence.

The drakes were quick to accept the saurian alliance - I suppose it's justified because they are in an unfamiliar land and they are weary of their travel and fighting (plus the saurians were quite subservient). I understand complementing the lawful drakes with the chaotic saurians is for game-play balance (especially with the cave scenario), but I just find it quite in contrast to, for example, Konrad or Haldric stumbling on to drake territory and attempting to make peace before the drakes quickly attack (granted, in both cases, the drakes are established in the land, not seeking refuge like Galun).

The drakes are often depicted as being war-like - does this include fighting against their own kind? I know the fear in this campaign is that soon drake numbers will overrun the land and drake will have to fight drake. I just found Vank being sad to 'fight our own kind' to be unusual - though I suppose he's been a moderating influence on Galun's more aggressive mentality.

Overall, I think the campaign suits its intermediate rating. I also think it is suitable for inclusion as a new campaign.

CelticMinstrel commented:
> plus the saurians were quite subservient

Hm? Don't the saurians have an actual empire either currently or in the recent past? Does it make sense for them to be so subservient?

Wedge009 commented:
I think it's fine for the sake of game-play - one could argue it's only a small group or one tribe. It just felt a bit odd to me for the we're-so-superior drakes to accept the saurians so quickly.

CelticMinstrel commented:
Perhaps they feel greater kinship to the reptilian saurians than they do towards humanoid races.

sigurdfdragon commented:
> This doesn't make sense to me - I would write 'flew back to Morogor' or 'returned to Morogor'.

Will do, thanks

> I found a problem too: if Galun's killing blow leading to victory and the end of the scenario is what brings him to the next level, the scenario is still treated as a failure because the level advancement occurs after the ending dialogue. Not sure if there's a way around that, it might be limited by the way the engine orders events.

I currently have the check as part of last breath, moving it to enemies defeated should fix it.

> How did humans come to be on Morogor? Were they left behind during the flight from the Green Isle? Escapees, deserters and the like?

Intended to be TRoW type Wesfolk from the green isle that got lost at sea/failed expeditions

> I am confused what is meant by short vs long pig. I suppose long pig means any humanoid race, not necessarily humans? (Small long pig = dwarf)

Yes

> How does the term runner fit into this?

It's their term for escaped slaves hunted as game.


>  I find it curious that there's no dialogue at all between the drakes and the humans/elves. While I suppose humans would be quick to attack anything they feared, I would have thought the elves would make more of an attempt to communicate before resorting to violence.

There could be, I really just haven't gotten around to putting some in.

> I am wondering, though, what is the reason for Galun to outlaw slavery? Is it to minimise the drain on the drakes' food supply? It's not clear to me, if the drakes keep humans for food, do they have to feed them as well?

I'm going to drop the whole slavery thread for 0.9.7 to have a better focus on the morality of apex predators, and their adaptation to the reality of better armed and organized long pigs on the great continent.

> Also: why would they now choose to avoid confrontation with the other sapient races? (They are 'superior', etc.)

Planning on changing one of the edicts to: "Avoidance of dangerous game and respecting them as fellow predators" to help with that. 


> Hm? Don't the saurians have an actual empire either currently or in the recent past? Does it make sense for them to be so subservient?

The saurian empire is still in existence, it's destroyed in LoW, and IIRC, it's in the location of the Swamp of Dread roughly.
The ones in WoV are subservient, because they are remnants of the followers of The Dragon from TRoW, the dragon having been slain by Haldric. They see Galun and his flight, and think it a good replacement for the dragon.

> I think it's fine for the sake of game-play - one could argue it's only a small group or one tribe. It just felt a bit odd to me for the we're-so-superior drakes to accept the saurians so quickly.

Noted, one of the lines I'm adding to 07 Elensefar about the long pigs here being more dangerous should help with this

sigurdfdragon commented:
Updated the PR with version 0.9.7.
See https://github.com/sigurdfdragon/Wings_of_Victory/commits/1.15 for the changes

CelticMinstrel commented:
Other than that I kinda feel it would be cooler to make up a word rather than using "pig", those changes are a definite improvement. (Note: If you _do_ make up a word, be sure to check the information on the phonology of the Drake language.)

Wedge009 commented:
> Intended to be TRoW type Wesfolk from the green isle that got lost at sea/failed expeditions

That wasn't clear to me, but I suppose it does make sense. I'm not sure if it's possible to sneak in a reference to them being Wesfolk without being obviously expository.

> It's their term for escaped slaves hunted as game.

I see. I think removing the 'runner' terminology as you did probably helps simplify things.

> There could be, I really just haven't gotten around to putting some in.

I suppose it's fine as-is, but I think having some dialogue could improve things.

> I'm going to drop the whole slavery thread for 0.9.7 to have a better focus on the morality of apex predators, and their adaptation to the reality of better armed and organized long pigs on the great continent.

Sounds good.

> Planning on changing one of the edicts to: "Avoidance of dangerous game and respecting them as fellow predators" to help with that.

Interesting. A bit of humility for the 'mighty' drakes. I suppose one has to swallow pride when facing possible extinction.

> The ones in WoV are subservient, because they are remnants of the followers of The Dragon from TRoW, the dragon having been slain by Haldric. They see Galun and his flight, and think it a good replacement for the dragon.

I forgot about the saurians being allies of the dragon - though I do remember a dragon as being one of the four tasks Haldric has to do for the elves as a condition for being able to settle and establish Wesnoth. Nice link, I suppose this makes sense.

> Noted, one of the lines I'm adding to 07 Elensefar about the long pigs here being more dangerous should help with this

Sounds good.

I agree that something better than 'pig' could be used, but I wouldn't know what to use. I don't think I have any major outstanding concerns at this point.

CelticMinstrel commented:
For the record, I was thinking just literally making up a random word that sounds vaguely similar to the drake names (using the same phonotactical rules), and using that instead of "pig" everywhere.

Wedge009 commented:
True. I've seen many a time in fiction where, for example, a made-up swear word is used (often even rendered in italics) and the reader might not know the exact meaning or intention of the word but it's very clear that it's a swear word, insult, etc. Obviously used to get stuff past the censors.

sigurdfdragon commented:
Thanks for the feedback.

> I kinda feel it would be cooler to make up a word rather than using "pig"

> I agree that something better than 'pig' could be used

I wasn't too thrilled with how it flowed as well. Will replace it in the next version. Possibly repurposing 'runners' to refer to 'standing upright game' as opposed to 'short pigs' that gait on all fours. We'll see.

> I suppose it's fine as-is, but I think having some dialogue could improve things.

I'll see what I can come up with.

sigurdfdragon commented:
Updated the PR with version 0.9.8.
See https://github.com/sigurdfdragon/Wings_of_Victory/commits/1.15 for the changes

Notable Changes:
Replaced all occurrences of pig with prey or game (ie, short prey, long prey, prey-speech, ect.)
Added some addition dialog in S07 Wesmere
Some other clarifications/cleanups

Unless there are any further significant issues, I'll be moving toward final touchups & merging the PR on the 27 or 28th of April

CelticMinstrel commented:
Not entirely happy with "prey" either, but it's worlds better than "pig", so I suppose I can live with it.

Wedge009 commented:
One thing it does do is help reinforce the idea that drakes consider themselves as superior as a race and reduces the emphasis on them eating other races. I suppose there might be a better option but this is a positive change to me.

AI0867 commented:
> I am confused what is meant by short vs long pig.

"Long pig" is a common euphemism for human meat, due to reported similarities. "Short pigs" would be the regular ones. The drakes seem to use "long pig" to refer to any sentient prey, with "short long pig" then logically being dwarves.

Whether drakes would use such a euphemism for sentient prey is a different question (human slaves could be a reason), but "long prey" seems to have lost all of the original meaning.

Wedge009 commented:
> "Long pig" is a common euphemism for human meat, due to reported similarities.
Are we talking about real-world or Wesnoth terminology? I'm aware pigs are often used as human analogues for scientific study but I don't see how this applies to the Wesnoth universe. I worked out that 'pig' referred to the various 'lesser races' but I felt the dialogue was perhaps trying to attach too much meaning to such an analogy. At any rate, 'pig' no longer seems to be in use as of the current revision.

AI0867 commented:
> > "Long pig" is a common euphemism for human meat, due to reported similarities.
> 
> Are we talking about real-world or Wesnoth terminology?

This is real-world terminology of cannibals.

> I worked out that 'pig' referred to the various 'lesser races' but I felt the dialogue was perhaps trying to attach too much meaning to such an analogy.

The real-world terminology does not have any "lesser race" implications AFAIK. It is simply a fairly good description that won't arouse too much suspicion.

 > At any rate, 'pig' no longer seems to be in use as of the current revision.

As I said, the phrasing is pretty meaningless. "Long pig" meant something, even if it is fairly obscure. "Long prey" is meaningless and causes awkward dialog. I'd sooner use "thinking prey" or "smart prey" or the like.

Wedge009 commented:
I don't know any cannibals, so this was completely new to me.

I only meant that the way the drakes used the term implied that they saw themselves as 'superior', which I understand was sigurd's intent.

Perhaps it is only meaningful to someone fluent in cannibal terminology.

AI0867 commented:
> I don't know any cannibals, so this was completely new to me.
> 
> Perhaps it is only meaningful to someone fluent in cannibal terminology.

Well, it would hardly be the first obscure reference in wesnoth lore.
 
> I only meant that the way the drakes used the term implied that they saw themselves as 'superior', which I understand was sigurd's intent.

IIRC, the use of the "long pig" terminology comes from ESR's lorecrafting efforts, but I'm not sure if it's a good choice. I'd suggest dropping the "long" adjective and designing some new term that makes more sense in this context. It's not worth keeping just for the "short long pig" joke.

sigurdfdragon commented:
Thanks for the recent feedback.

I'm postponing merging until I get the short/long/pig/prey/runner/game thing sorted.

sigurdfdragon commented:
Have cleaned up the short/long/pig/prey/runner/game thing.

Changes are here: https://github.com/sigurdfdragon/Wings_of_Victory/commit/a40062cc8ae1c928f01dd11fa978275a154ee6ad

Hopefully, that's it. Aiming to merge pr on Wednesday, May 1st

Wedge009 commented:
I think this is a suitable compromise - I know many people have differing opinions. The campaign as a whole is okay to me, I'm comfortable with this being merged.

CelticMinstrel commented:
I think the plan for 1.15 will be decided upon after this is merged, actually? We may actually get a 1.15 release soonish, in other words.

CelticMinstrel commented:
I'll add that "trotters" might work better than "gallopers", as "gallop" is pretty horse-specific if I'm not mistaken.

Wedge009 commented:
That could work, I suppose. Horses move from walking to trotting to galloping - I don't know of cases where gallop is used for anything besides horses. If I recall correctly Wesnoth sentient races are primarily bipedal and the only quadrupeds we see are the mounts: horses, wolves, etc.

AI0867 commented:
> I'll add that "trotters" might work better than "gallopers", as "gallop" is pretty horse-specific if I'm not mistaken.

https://en.wikipedia.org/wiki/Canine_gait
https://en.wikipedia.org/wiki/Gait

CelticMinstrel commented:
I see. That said, I also think "trotters" sounds better.

sigurdfdragon commented:
Updated the PR with version 1.0.0 & merged
See https://github.com/sigurdfdragon/Wings_of_Victory/commits/1.15 for the changes

Now that the campaign has been merged, feedback is still welcome, but I'll be taking a longer-term approach to further updates unless they are critical bugfixes.
Forum thread: https://forums.wesnoth.org/viewtopic.php?f=8&t=49140
There may be 1-3? minor sets of updates before the end of the 1.15 cycle

Thanks to:
nemaara, CelticMinstrel, jostephd, shikadiqueen, slavrenyuk, Pentarctagon, Wedge009, AI0867
for all the feedback, you helped make this a better campaign.
